src/Sprites/ArtFile.h

```diff
@@ -1,23 +1,17 @@
 #pragma once
 
 #include "../Rect.h"
-#include <array>
+#include "ImageType.h"
+#include "Color.h"
 #include <vector>
 #include <cstdint>
 
-struct Color {
-	uint8_t red;
-	uint8_t green;
-	uint8_t blue;
-	uint8_t alpha;
-};
-
 struct ImageMeta {
 	uint32_t scanlineByteWidth; //number of bytes in each scanline of image (this should be the width of the image rounded up to a 32 bit boundary)
 	uint32_t pixelDataOffset; // Offset of the pixel data in the .bmp file
 	uint32_t height; // Height of image in pixels
 	uint32_t width; // Width of image in pixels
-	uint16_t type;
+	ImageType type;
```

Comment:
Hmm, should we be explicit about the size of `ImageType` here? All the other fields are quite explicit. As it stands now, the size of this struct is dependent on external information, namely the base type of `ImageType`. We could potentially write:
```cpp
ImageType type:16;
```

I think that would make the serialization code a little more robust in regards to any changes to the ImageType enum.

Reply:
I think this is a mute point as ImageType should actually be a bitfield as mentioned earlier. I'll update shortly.

I made ImageType a uint16_t scoped enum, so it should have always been the same size as uint16_t.

Reply:
Hmm, very good point. I didn't realize I was commenting on the same field there.

As for the `uint16_t` base type on the enum, the part I didn't like so much about it is the base type is specified with the enum declaration, separate from the struct declaration. It's the struct declaration where the size really matters, and the other fields all had explicit sizes in the struct. I suppose if the enum is changed to a struct of bitfields, it's the same issue, where the size of the sub-component is controlled by the separate declaration of that subcomponent.

I suppose the other option is to embed the bitfield struct within the `ImageMeta` namespace, or embed the fields themselves, into the `ImageMeta` struct. Not sure what is better. I suppose it depends on if those bit fields have use outside the context of the `ImageMeta` struct.